### DISCUSSION TITLE
[Feature Request] cursorWordPartLeftSelect + cursorWordPartRightSelect

### DISCUSSION THREAD

**Discussion by dominictobias** [zed-industries/zed#20670] [Feature Requests]
VSCode calls these shortcuts `cursorWordPartLeftSelect` and `cursorWordPartRightSelect` and I also used it in Sublime. I miss it in Zed.

What it does is select up to the next casing change:

![Screenshot 2024-11-14 at 21 51 57](https://github.com/user-attachments/assets/8c7534cc-0d80-4b56-ad8e-697d14b362c7)

I think; VSCode seems to mess up in certain case changes like here it doesn't select the "E" which seems like a bug to me. But not an issue in normal code:

![Screenshot 2024-11-14 at 21 56 49](https://github.com/user-attachments/assets/32cfbf23-587f-4c5e-8406-c6310d980cf4)

It should also stop when hitting non 0-9|a-z|A-Z:

![Screenshot 2024-11-14 at 21 58 24](https://github.com/user-attachments/assets/af4b67a5-ea42-4840-bcbd-7adb9ccb7863)
![Screenshot 2024-11-14 at 21 58 08](https://github.com/user-attachments/assets/893a1b2d-54aa-465f-b11f-134fd1d75f17)

I wonder if you'd consider this feature in Zed? Thanks

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
